Fix patches for cuda 12.9

hmaarrfk opened the pull request:
Trying to address https://github.com/conda-forge/tensorflow-feedstock/pull/467#discussion_r2575467071

but will have to verify that this is working by uploading to my channel.

We have an internal test that seems to be failing with the latest 12.9 build.
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

hmaarrfk committed:
Fix patches for cuda 12.9

h-vetinari committed:
renormalize patches

hmaarrfk commented on recipe/patches/0044-Support-for-TFv2.20-to-compile-with-CUDA-v12.9.1.patch:
```diff
@@ -0,0 +1,39 @@
+diff --git a/tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc b/tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc
+index ae3fe6d1..737e22e5 100644
+--- a/tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc
++++ b/tensorflow/core/kernels/dynamic_partition_op_gpu.cu.cc
```
tbh, even the `.diff` didn't work, and I had to manually write this, by copy pasting lines and removing the `+` and `-`.

The attribution is there, i'm not taking it, I included a reference to the PR.

h-vetinari replied:
> it was difficult to backport the patch with `git am`

It's easier to take the local branch (as of what's currently on `main`; if you need to recreate the branch you can do `find ../path/to/feedstock/recipe/patches/ | xargs git am` after `git checkout tags/v2.19.1 -b conda`, for example), and then rebase using `git rebase -i tags/v2.19.1`, whereupon you insert the `pick 6014ea08c6171c62c2a12956a3cf2162be0b6836` for the commit https://github.com/tensorflow/tensorflow/commit/6014ea08c6171c62c2a12956a3cf2162be0b6836 we want to backport.

Normally, that's easy enough to fix any eventual conflicts. In this case, https://github.com/tensorflow/tensorflow/pull/99364 is a bit special, because it wasn't squash-merged; originally I had backported https://github.com/tensorflow/tensorflow/commit/c59bd3fdce713f3427ddbfe2dc9fbb01490d0bdc (which was later fixed-up before merging), but now, after the new commit, it was easier to backport https://github.com/tensorflow/tensorflow/commit/e50c8eb3795972dbb6b0dc570bd61daeedc1fade with `git cherry-pick e50c8eb3795972dbb6b0dc570bd61daeedc1fade -m 1` on top of the backport of https://github.com/tensorflow/tensorflow/commit/6014ea08c6171c62c2a12956a3cf2162be0b6836. I still kept the original attribution though, because the attribution of the merge commit is uninteresting. It's a bit of a special case, as I said.

hmaarrfk committed:
Update ml-dtypes

hmaarrfk committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.01.11.54.38

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.9

hmaarrfk committed:
Bump build number for myself

hmaarrfk committed:
Oops compile with cuda only...

hmaarrfk committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.01.11.54.38

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.9

hmaarrfk committed:
Only build python 3.12

isuruf committed:
Update skip conditions
